shared/node.go

```diff
@@ -53,15 +53,13 @@ func Create(cfg *genesis.Config, genesis *genesis.GenesisState) (n *Node, err er
 	if err != nil {
 		return nil, err
 	}
-
-	pk, err := cryptoPocket.NewPrivateKey(cfg.Base.PrivateKey)
+	addr, err := p2pMod.GetAddress()
```

Comment:
Could we get this from the config instead?

Reply:
Likely, we can convert the private key string into an object and then hash and convert to an address from the configuration or just use the function that is already exposed by P2P module. 

AFAIK the address doesn't even need to be in the 'node'

Reply:
`AFAIK the address doesn't even need to be in the 'node'`

My thinking is currently this:
1. Getting the address is something every module is likely to have to do multiple times
2. We want to avoid doing the hash + conversion more than once
3. What should we do given the architecture of different modules having different keys, and in turn, addresses?

AI: Should we just make this easily accessible via the node?

Reply:
Leaving this discussion for the future.